src/ServiceBus.AttachmentPlugin/AzureStorageAttachmentConfiguration.cs

```diff
@@ -10,16 +10,23 @@ public class AzureStorageAttachmentConfiguration
         /// <param name="connectionString"></param>
         /// <param name="containerName"></param>
         /// <param name="messagePropertyToIdentifyAttachmentBlob"></param>
+        /// <param name="messagePropertyToIdentifySasUri">The message user property to use for the generated sas uri.</param>
+        /// <param name="sasTokenValidInSeconds">In seconds, how long the generated sas uri should be valid.</param>
         /// <param name="messageMaxSizeReachedCriteria">Default is always use attachments</param>
         public AzureStorageAttachmentConfiguration(string connectionString,
             string containerName = "attachments",
             string messagePropertyToIdentifyAttachmentBlob = "$attachment.blob",
+            string messagePropertyToIdentifySasUri = "$attachment.sas.uri",
+            long sasTokenValidInSeconds = 0,
```

Comment:
Since `TokenGenerator.GetBlobSasUri()` take valid time as `TimeSpan`, you could make this argument timespan as well.
It would also enforce more descriptive API when someone uses it as they'd be expressing it with `TimeSpan.FromDays(1)` rather than 60\*60\*24 🙂